src/OpenApi/src/Transformers/ActivatedOpenApiDocumentTransformer.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal sealed class ActivatedOpenApiDocumentTransformer(Type transformerType) : IOpenApiDocumentTransformer
+{
+    internal IOpenApiDocumentTransformer? Transformer { get; set; }
+
+    internal void Initialize(IServiceProvider serviceProvider)
+    {
+        Transformer ??= ActivatorUtilities.CreateInstance(serviceProvider, transformerType) as IOpenApiDocumentTransformer;
```

Comment:
Should this be a normal cast?  Otherwise, it'll be activated over and over again if it fails to satisfy the interface.  Alternatively, we could probably test the type in the ctor.

Reply:
We have a generic constraint in the `UseTransformer` call that registers this type so I think that provides a reasonable guarantee. I did add a `Debug.Assert` to validate the type since if a type that doesn't implement `IOpenApiDocumentTransformer` is passed its likely an issue with our own code.

Reply:
Change the `as` to a cast then?